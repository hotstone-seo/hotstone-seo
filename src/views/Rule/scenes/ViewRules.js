import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { Button } from 'antd';
import { fetchRules } from 'api/rule';
import { RuleList } from 'components/Rule';

function ViewRules({ match }) {
  const [rules, setRules] = useState([]);

  useEffect(() => {
    let _isCancelled = false;
    fetchRules()
      .then(rules => {
        if (!_isCancelled) {
          setRules(rules);
        }
      });

    return () => {
      _isCancelled = true;
    };
  })

  return (
    <div>
      <Button
        type="primary"
        style={{ marginBottom: 16 }}
      >
        <Link to={`${match.url}/new`}>Add new Rule</Link>
      </Button>
      <RuleList rules={rules} />
    </div>
  )
}

export default ViewRules;
