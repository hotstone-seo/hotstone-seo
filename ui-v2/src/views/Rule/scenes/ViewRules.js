import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Link, useHistory } from 'react-router-dom';
import { PageHeader, Button, message } from 'antd';
import { fetchRules, deleteRule } from 'api/rule';
import { RuleListV2 } from 'components/Rule';

function ViewRules({ match }) {
  const [rules, setRules] = useState([]);

  useEffect(() => {
    let isCancelled = false;
    fetchRules().then((newRules) => {
      if (!isCancelled) {
        setRules(newRules);
      }
    });

    return () => {
      isCancelled = true;
    };
  }, []);

  const history = useHistory();

  const showEditForm = (rule) => {
    history.push(`${match.url}/${rule.id}`);
  };

  const handleDelete = (rule) => {
    deleteRule(rule.id)
      .then(() => {
        message.success(`Successfully deleted ${rule.name}`);
        setRules(rules.filter((item) => item.id !== rule.id));
      })
      .catch((error) => {
        message.error(error.message);
      });
  };

  return (
    <div>
      <PageHeader
        title="Rules"
        subTitle="Manage Tags on matching URL Rules"
        style={{ background: '#fff' }}
      />
      <div style={{ padding: 24 }}>
        <Button type="primary" style={{ marginBottom: 16 }}>
          <Link to={`${match.url}/new`}>Add new Rule</Link>
        </Button>
        <RuleListV2
          rules={rules}
          onClick={showEditForm}
          onEdit={showEditForm}
          onDelete={handleDelete}
        />
      </div>
    </div>
  );
}

ViewRules.propTypes = {
  match: PropTypes.shape({
    url: PropTypes.string,
  }).isRequired,
};

export default ViewRules;
