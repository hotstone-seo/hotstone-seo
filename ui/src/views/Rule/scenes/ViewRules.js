import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import { PageHeader, Button, message } from 'antd';
import { deleteRule, updateStatusStartRule } from 'api/rule';
import { RuleListV2 } from 'components/Rule';
import { PlusOutlined } from '@ant-design/icons';

function ViewRules({ match }) {
  const history = useHistory();
  const [listRule, setListRule] = useState([]);

  const showEditForm = (rule) => {
    history.push(`${match.url}/${rule.id}`);
  };

  const handleDelete = (rule) => {
    deleteRule(rule.id)
      .then(() => {
        message.success(`Successfully deleted ${rule.name}`);
        setListRule(listRule.filter((item) => item.id !== rule.id));
      })
      .catch((error) => {
        message.error(error.message);
      });
  };

  const handleUpdateStatusStart = (checked, rule) => {
    const onOFF = checked === true ? 'START' : 'STOP';
    /* updateStatusStartRule(rule.id)
      .then(() => {

        setListRule(listRule.filter((item) => item.id !== rule.id));
      })
      .catch((error) => {
        message.error(error.message);
      }); */
    message.success(`Successfully switch ${rule.name} to be ${onOFF}`);
    // console.log(`switch to ${checked}`);
  };

  const addDataRule = () => {
    history.push(`${match.url}/new`);
  };

  return (
    <div>
      <PageHeader
        title="Rules"
        subTitle="Manage Tags on matching URL Rules"
        style={{ background: '#fff' }}
      />
      <div style={{ padding: 24 }}>
        <Button
          data-testid="btn-new-rule"
          type="primary"
          style={{ marginBottom: 16 }}
          icon={<PlusOutlined />}
          onClick={() => addDataRule()}
        >
          Add New Rule
        </Button>
        <RuleListV2
          onClick={showEditForm}
          onEdit={showEditForm}
          onDelete={handleDelete}
          listRule={listRule}
          setListRule={setListRule}
          onChangeToggleButton={handleUpdateStatusStart}
        />
      </div>
    </div>
  );
}

ViewRules.propTypes = {
  match: PropTypes.shape({
    url: PropTypes.string,
  }).isRequired,
};

export default ViewRules;
