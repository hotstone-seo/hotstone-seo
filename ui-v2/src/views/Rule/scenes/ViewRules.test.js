import React from 'react';
import {
  render, wait,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import mockAxios from 'jest-mock-axios';

import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ViewRules from './ViewRules';

configure({ adapter: new Adapter() });

const respMock = [{
  id: 1, name: 'Airport Rule', url: '/foo-ds', url_pattern: '/airport', updated_at: new Date().toISOString(), created_at: new Date().toISOString(),
}];

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));


test('View Rules unit test', () => {
  const props = {
    match: { url: 'tes.com' },
  };
  const tree = mount(<ViewRules {...props} />);
  expect(tree).toMatchSnapshot();
});

/*
describe('View Rules unit test', () => {
  test('Rule List', async () => {
    const url = '/rules';
    const {
      queryByText,
    } = render(<ViewRules match={{ url }} />);

    await wait(() => {


      expect(mockAxios.get).toHaveBeenNthCalledWith(2);
      // expect(mockAxios.get).toHaveBeenCalledWith('/rules');
      mockAxios.mockResponse({ data: respMock });

      expect(queryByText(/Rules/)).toBeInTheDocument();
      expect(queryByText(/Airport Rule/)).toBeInTheDocument();
    });
  });
});
*/
