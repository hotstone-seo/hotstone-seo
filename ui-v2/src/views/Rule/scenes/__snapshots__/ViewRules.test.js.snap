// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Rules unit test 1`] = `ReactWrapper {}`;
